src/Experiment/src/AssemblyBuilder.cs

```diff
@@ -1,27 +1,68 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-﻿using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using System.Reflection;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
 
 namespace System.Reflection.Emit.Experimental
 {
+
     public class AssemblyBuilder: System.Reflection.Assembly
     {
-        internal AssemblyBuilder() { }
+        private static readonly Guid s_guid = Guid.NewGuid();
+        private BlobBuilder _emptyBlob = new BlobBuilder();//For later
+        private AssemblyName? _assemblyName;
+        private MetadataBuilder _metadata = new MetadataBuilder();
+        private IDictionary<string,ModuleBuilder> _moduleStorage = new Dictionary<string, ModuleBuilder>();
+
+        public AssemblyBuilder() { }
+        private AssemblyBuilder(AssemblyName name) 
+        {
+            _assemblyName = name;
+        }
 
         public void Save(string assemblyFileName)
         {
-            //New method
+            if (assemblyFileName == null)
+            {
+                throw new ArgumentNullException();
+            }
+            if (_assemblyName==null||_assemblyName.Name==null)
+            {
+                throw new ArgumentNullException();
+            }
+            //Add assembly metadata
+            _metadata.AddAssembly(//Metadata is added for the new assembly - Current design - metdata generated only when Save method is called.
+               _metadata.GetOrAddString(value: _assemblyName.Name),
+               version: _assemblyName.Version ?? new Version(1, 0, 0, 0),
```

Comment:
.NET Framework defaults to `0.0.0.0` when a version isn't specified, so we should match that behavior here.

Reply:
Also, Culture defaults to `neutral` and PublicKeyToken to `null`